### DISCUSSION TITLE
Sharing information about Fysetc Raspberry Pi 5 inch DSI Display

### DISCUSSION THREAD

**Discussion by iamhermes** [KlipperScreen/KlipperScreen#1491] [Show and tell]
Hello,

I want others to tell, the touchscreen will only work if you use an older raspbian image for pi. (connected to raspberry pi 3 b+)
I can't get the touch functional on latest Ubuntu or raspbian.
After contacting Fysetc they send me two firmware images (raspbian 11, desktop and a voron one) in which the touch works properly. And the bad: apt upgrade will stop the touchscreen to work.

I did some exploration about the issue and solved it by blocking package firmware-brcm80211 (apt-mark hold) to update. An update to latest available version 20230210-5~bpo11+1+rpt2 will break the touch. I think every raspbian image before February 2023 will work with this type of touch without updating. 

> pi@raspberrypi:~ $ dpkg -l |grep brcm
> hi  firmware-brcm80211                   1:20221012-1~bpo11+1+rpt1

> config.txt:
> #Enable DRM VC4 V3D driver
> #dtoverlay=vc4-kms-v3d
> #max_framebuffers=2
> dtoverlay=touchscreen-inverted-x=1,touchscreen-inverted-y=1
> display_lcd_rotate=2
> #Disable compensation for displays with overscan
> disable_overscan=1

Photo from touch IC:
![Screenshot_20241024-092442](https://github.com/user-attachments/assets/34f63cc9-995a-4395-8213-d963e30f0af4)
Screen:
![Screenshot_20241024-093107](https://github.com/user-attachments/assets/47696afd-571d-47d7-ae7c-12f4b97413c1)
https://www.fysetc.com/en-de/products/fysetc-raspberry-pi-5-inch-dsi-touch-screen-mipi-lcd-v1-1-5-inch-lcd-touch-screen-800-x-480-hd-hdmi-working-with-raspberry-pi5-4b-cm4-3b-3a-3b-cm3-cm3-2b-b-a

**Comment by alfrix** [comment]
it seems odd that the wireless firmware causes the touchscreen to stop working
so if you do `apt list --upgradable` that's the only package that remains ??

**Reply by iamhermes** [reply]
In the images provided by fysetc, 
It is the only package that destroys the function of the touchscreen after an apt upgrade. 

After the update it behaves exactly the same as when I use the last versions of raspbian and ubuntu (x32 or x64 makes no difference). 
The touch then reacts differently depending on the dtoverlay specification. 
The touch itself is recognized, but no event is generated. 
In the second case, the touch is recognized and generates events, but these are totally incorrect (the coordinates are outside the min/max with random numbers) and involuntary detections are displayed with random values ​​also outside the permitted range.
Or the touch device is not detected.
